### ISSUE TITLE
Adjust for pandas 2.1.0

### ISSUE THREAD

**Issue by khaeru** [IAMconsortium/pyam#780]
Using pyam-iamc v1.9.0 with pandas 2.1.0 appears (initially) to work alright, at least for the subset of functionality used in iiasa/message_data.

However, there are (many) FutureWarnings emitted. Here are the ones I've observed:
```
pyam/aggregation.py:210: FutureWarning: The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass 'sum' instead.

pyam/aggregation.py:210: FutureWarning: The provided callable <function sum at 0x7fbbe5333eb0> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass 'sum' instead.

pyam/core.py:2691: FutureWarning: unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.

pyam/plotting.py:466: FutureWarning: The behavior of array concatenation with empty entries is deprecated. In a future version, this will no longer exclude empty items when determining the result dtype. To retain the old behavior, exclude the empty entries before the concat operation.

pyam/plotting.py:485: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
```

Again, AFAICT there's no bug, and we will use [`filterwarnings()`](https://docs.python.org/3/library/warnings.html#warnings.filterwarnings) for now. This issue is just to track when those filters can be removed.

**Comment by danielhuppmann** [comment]
Thanks @khaeru, but these warnings should have been fixed already by #777 - can you confirm?

**Comment by khaeru** [comment]
Great, thanks! I will try to check against `main` in the next day or two.

**Comment by coroa** [comment]
Note that there seems to be at least one non-trivial bug with the combination of pandas 2.1 and pyam 1.9:

```
import pandas as pd
import pyam

test_df = (
    pd.DataFrame(
        dict(
            model="any",
            scenario="any",
            region=["AUS", "USA"],
            variable="any",
            unit="any",
            year=2000,
            value=50,
        )
    )
    .set_index(["model", "scenario", "region", "variable", "unit", "year"])
    .value.unstack("year")
)
 
pyam.IamDataFrame(test_df.iloc[[0]]).region
/Users/coroa/repos/pyam/pyam/core.py:2691: FutureWarning: unique with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
  index = pd.unique(index)
```
will result in `["AUS", "USA"]` as regions rather than only `["AUS"]` since we only included the first line.

It is fixed on the main branch/the release candidate.

**Comment by danielhuppmann** [comment]
Re @coroa, indeed, that was noted and fixed in #763 (and a test added to guard against regression) - but there isn't really anything we can do about past releases...

**Comment by danielhuppmann** [comment]
Closing this issue, please reopen if you encounter any unfixed future-warnings.
